File: algorithms/203.swift
// https://leetcode.com/problems/remove-linked-list-elements

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */

class Solution {
    func removeElements(head: ListNode?, _ val: Int) -> ListNode? {
        guard let h = head else { return nil }

        var current = h
        var next = h.next
        while let n = next {
            if n.val == val {
                current.next = n.next
            } else {
                current = n
            }
            next = n.next
        }
        if h.val == val { return h.next }
        else { return h }
    }
}

File: algorithms/203.py
# https://leetcode.com/problems/remove-linked-list-elements


# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if head is None:
            return None

        current = head
        next = head.next
        while next is not None:
            if next.val == val:
                current.next = next.next
            else:
                current = next
            next = next.next

        return head if head.val != val else head.next
